test/timeout.js

```diff
@@ -70,6 +70,32 @@ tape('job timeout', function (t) {
   q.start()
 })
 
+tape('job-based opt-out of timeout', function (t) {
+  t.plan(1)
+
+  var timeouts = 0
+  var q = queue({ timeout: 5 })
+  function wontTimeout (cb) {
+    setTimeout(cb, 8)
+  }
+
+  wontTimeout.timeout = undefined
```

Comment:
although undefined, false etc should work, i think `null` communicates intent most clearly, what do you think?